src/System.Private.CoreLib/shared/System/IO/BinaryReader.cs

```diff
@@ -363,23 +362,15 @@ private int InternalReadChars(Span<char> buffer)
         {
             Debug.Assert(!_disposed);
 
-            int numBytes = 0;
-            int index = 0;
-            int charsRemaining = buffer.Length;
+            int totalCharsRead = 0;
 
-            if (_charBytes == null)
+            while (!buffer.IsEmpty)
             {
-                _charBytes = new byte[MaxCharBytesSize];
-            }
+                int numBytes = buffer.Length;
```

Comment:
What you have is fine; just wondering if this would be any better as:
```C#
while (true)
{
    int numBytes = buffer.Length;
    if (numBytes == 0)
    {
        break;
    }

    ...
```

Reply:
It looks about the same to me.